[release notes] Release notes for ZLIB-CF (#3970)
root-project/root: README/ReleaseNotes/v618/index.md

* The deprecrated `I/O` plugins for  `HDFS`, `Castor` and `RFIO` have been removed.

> * A new version of zlib (ZLIB-CF), based on ZLIB Cloudlare fork and CMSSW zlib Cloudlare fork, was added in ROOT.

Reviewer: ```suggestion
* A new version of zlib, based on CMS's zlib Cloudflare fork, was added to ROOT
```

> It is a high performance modification of zlib 1.2.8 compression algorithm, which offers up to 70% faster compression speed for Intel/ARMv8 processors  for tested use cases (30% - for fast compression (`CompressionSettings(101)`) and 70% for the slow compression (`CompressionSettings(109)`)), using SIMD/NEON acceleration for CRC32 and Adler computations. ZLIB-CF was adopted in ROOT using a zlib builtin mechanism. To enable ZLIB-CF, please use `-Dbuiltin_zlib=ON` flag.

Reviewer: ```suggestion
It is a high-performance modification of the zlib 1.2.8 compression algorithm, offering up to 70% faster compression speed for Intel/ARMv8 processors for tested use cases (30% - for fast compression (`CompressionSettings(101)`) and 70% for the slow compression (`CompressionSettings(109)`)), using SIMD/NEON acceleration for CRC32 and Adler computations. It is available in ROOT as zlib builtin: to enable it, please configure with `-Dbuiltin_zlib=ON`.
```
